subprojects/dependency-management/src/main/java/org/gradle/internal/component/model/LocalComponentDependencyMetadata.java

```diff
@@ -107,7 +107,7 @@ public String getDependencyConfiguration() {
 
     @Override
     public Set<ConfigurationMetadata> selectConfigurations(ComponentResolveMetadata fromComponent, ConfigurationMetadata fromConfiguration, ComponentResolveMetadata targetComponent, AttributesSchema attributesSchema) {
-        assert fromConfiguration.getHierarchy().contains(getOrDefaultConfiguration(moduleConfiguration));
+        // assert fromConfiguration.getHierarchy().contains(getOrDefaultConfiguration(moduleConfiguration));
```

Comment:
I think it would be unfortunate if we started removing our assertions. We should just disable assertions in our performance tests.

Also, I'd be kinda surprised if hotspot wasn't smart enough to ignore disabled assertions when inlining.